### ISSUE TITLE
https everywhere and noScript hang Firefox nightly

### ISSUE THREAD

**Issue by yoshi-k** [EFForg/https-everywhere#1352]
I encountered a interaction between https-everywhere and noScript, which conspire to render Firefox nightly unusable when trying to play a youtube Video. ( I will also submit this bugreport to  noScript, no idea where to fix this issue. ) 

**Steps to reproduce**
* Visit http://www.sfsignal.com/archives/2015/04/trailer-park-ash-vs-evil-dead-poltergeist-pixels/
* Open sidebar ( CTRL-H ) it seems to be important that the Video container is smaller than the player, such that the controls are partially occluded. 
* Try to use the "Watch on youtube.com" button. 
* Result: Firefox hangs. 

Output of Firefox: https://gist.github.com/yoshi-k/2081ce69e7bd32a64413

**Versions:**
Firefox nightly 40.0a1
NoScript 2.6.9.20
HTTPS Everywhere  5.0.1
`$uname -a`
`Linux Angmar 3.19.2-1-ARCH #1 SMP PREEMPT Wed Mar 18 16:21:02 CET 2015 x86_64 GNU/Linux`

**Comment by aleksejrs** [comment]
Since approximately the week before the Firefox 37.0 release (add-ons update automatically), my Firefox (36.0, 37.0, 38.0b1) frequently janks for a few seconds, or maybe for minutes and sometimes hangs forever.  It usually happens when a URL is loading, especially one URL in a Private window with a simple Greasemonkey script changing a URL of an image, but also bugzilla.mozilla.org and other pages.  Sometimes, a busy script warning for chrome://noscript/content/ScriptSurrogate.js:323 appeared.

Disabling HTTPS Everywhere and NoScript seemed to fix the problem.  Disabling only one of them, not necessarily so.  Disabling other add-ons at least alleviated the problem, but I haven't found which ones were necessary for it to happen.

**Comment by aleksejrs** [comment]
Oops, I accidentally deleted my comment because of the stupid icon fonts that were blocked.

**Comment by aleksejrs** [comment]
If anyone has my comment, please post it.  Thanks!

**Comment by aleksejrs** [comment]
Since approximately the week before the Firefox 37.0 release (add-ons update automatically), my Firefox (36.0, 37.0, 38.0b1) frequently janks for a few seconds to apparently forever.  After I've deleted HTTPS Everywhere and NoScript, it still happened sometimes, maybe in fewer situations.  Debian testing.

**Comment by reedy** [comment]
It may be worth filing this upstream with Firefox too

**Comment by yoshi-k** [comment]
I looked a bit into it again and prompted by a reply from the NoScript devs have tried a few additonal things. The report at the NoScript forum can be found [here](https://forums.informaction.com/viewtopic.php?f=10&t=20684&p=75497#p75497) and is copied below: 

I looked a bit into it again and have now updated to
FF nightly 40.0a1 (2015-04-10)
NoScript 2.6.9.21

First of all the sfsignal link still crashes the browser ( sorry for the misinformation :(
That was related to e10, with e10 enabled it works. ( It does not with e10 switched off.)

It also works, if NoScript's Clear Click is switched off. Hopefully interestingly, if I turn Clear Click off and then on again, it works until the end of the session but reverts to the old behavior, that is FF hangs, after a FF restart. ( If Clear Click needs a browser restart, it does not display a warning in about:addons.)

The browser also crashes when all rules in HTTPS everywhere are turned off.

Additionally I tried to build a test case, based on the assumption that the issue may be related to a youtube iframe inside of an too small div container, but in that case I could only get a warning, apparently from NoScript.
Testpage:
https://gist.github.com/yoshi-k/c159db8c4a8b988f456a#file-testpage-html
http://bl.ocks.org/yoshi-k/c159db8c4a8b988f456a#file-index-html ( Turns out one can host gists at github.)
Error message (console output):
https://gist.github.com/yoshi-k/377a0c35753940e992df#file-output_ff_13-4-15-txt
